test/utils.test.sh

```diff
@@ -390,6 +390,60 @@ assert "contains" "${log}" "pushing test to us-east-1"
 assert "contains" "${log}" "found existing image for test in us-west-2, skipping push"
 assert "equal" "${FAILURE}" "" "should not have any failures"
 
+# bucket_push() no ImageBucketPrefix
+tag_test "bucket_push"
+export ImageBucketPrefix=""
+
+log=$(bucket_push)
+expected="nothing to do"
+assert "equal" "${log}" "${expected}"
+
+# bucket_push()
+export ImageBucketPrefix="something"
+export bucket_regions=(region1)
+export tmpdir="temp"
+export repo="repo"
+export after="1"
+
+function aws(){
+  if [ ${1} == "s3" ] && [ ${2} == "cp" ]; then 
+    assert "equal" "$*" "s3 cp temp/repo-1.tar.gz s3://something-region1/images/repo/1.tar.gz --only-show-errors" "calls aws s3 cp with correct vars" >&2
+  else 
+    FAILURE="should call aws s3 cp"
+  fi
+}
+
+log=$(bucket_push)
+assert "equal" "${FAILURE}" "" "should not have any failures"
```

Comment:
can this message be also assigned to a variable like other places in the code and than logging and asserting on that?